jsonpickle/util.py

```diff
@@ -43,10 +43,19 @@ def is_type(obj):
     >>> is_type(Klass)
     True
     """
+    # use "isinstance" and not "is" to allow for metaclasses
     if PY3:
-        return type(obj) is type
+        return isinstance(obj, type)
     else:
-        return type(obj) is type or type(obj) is types.ClassType
+        return isinstance(obj, (type, types.ClassType))
+
+
+def has_method(obj, name, *args, **kwargs):
+    try:
+        getattr(obj, name)(*args, **kwargs)
+        return True
+    except:
+        return False
```

Comment:
Thanks @aldanor , that makes sense.  I guess if we can get away with not calling the method then that's ideal, but if it doesn't work out then I'm okay being pragmatic.

RE: the dependency thing -- I'd like to avoid the dependency if possible. If you `git grep "if PY3" -- tests` you'll see a few cases where we skip tests based on the python version, e.g.:

        if PY3:
            self.skipTest('No classic classes in PY3')

Maybe the tests could only run if `enum` is importable -- that way it's a soft dependency and won't force users to install the `enum` module on py2.